### DISCUSSION TITLE
Generator already executing nav2 error

### DISCUSSION THREAD

**Discussion by chaidop** [gstavrinos/ai_msc_robotic_systems_assignments#48] [Q&A]
Hello from me as well,
I am very happy with the easy_mode first of all.
I have however encouneterd another issue nmot sure if it is the way I am implementing the code.
I go iteratively in a loop with goToPose and then I call navigator.goToPose(shelf_item_pose) while not navigator.isTaskComplete() to check the feedback. In the isTaskComplete, it gives ERROR:

```
 ValueError: generator already executing
```

Any ideas? Or hints? Or prayers?
I just want my robot to not get hurt :(

**Comment by gstavrinos** [comment]
This is a typical python error when a generator object is not correctly handled. Check the block and thread you are creating and initializing the object. This shouldn't be happening, and most likely is not a problem with the assignment. Konstantinos had the same issue reported in the comments here #44.
